Assets/Scripts/AI/Entities/GameEntity.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+using WorldAPI.Entities;
+using System;
+
+public abstract class GameEntity<T> : Entity<T> where T: GameEntity<T>
+{
+    public Transform TargetStructure;
+    public new readonly Dictionary<Material, int> Inventory = new Dictionary<Material, int>();
```

Comment:
In the future entities may hold other things then just "Materials". May want to consider changing this to something like "InventoryItem".